### PR TITLE
feat: support file upload in NodeJS

### PR DESCRIPTION
**Pull request by lynxtaa**
### What does this PR do?

Supports file upload in NodeJS using `client.request(...)`

### Related issues

https://github.com/nearform/graphql-hooks/issues/399

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

**Comment by mcollina** [comment]
I think it would be better if we could test that the stream data is actually receivied on the other end.

**Comment by lynxtaa** [comment]
@mcollina You're right but at the moment there are no tests of receiving data on the other end for any scenario. This kind of testing should be started from checking requests with JSON payloads (most common usage) and then file uploads (rarer case)... I think it's out of scope of this PR.

File: README.md [hunk 1]
```diff
@@ -140,6 +140,7 @@ const client = new GraphQLClient(config)
   - `getInitialState()`
   - See [graphql-hooks-memcache](packages/graphql-hooks-memcache) as a reference implementation
 - `fetch(url, options)`: Fetch implementation - defaults to the global `fetch` API. Check [Request interceptors](#request-interceptors) for more details how to manage `fetch`.
+- `FormData`: FormData implementation - defaults to the global `FormData` API.
```
Comment (bmullan91) [inline README.md]:
```suggestion
- `FormData`: FormData implementation - defaults to the global `FormData` API. Polyfill this in a node.js environment. See [file-uploads-nodejs](#file-uploads-nodejs) for more info.
```

File: README.md [hunk 2]
```diff
@@ -583,6 +584,31 @@ export default function PostForm() {
 }
 ```
 
+NodeJS example:
```
Comment (bmullan91) [inline README.md]:
```suggestion
### File uploads Node.js
```

@lynxtaa Double check the link works to this header - we may need to drop the `.` in Node.js

File: README.md [hunk 3]
```diff
@@ -583,6 +584,31 @@ export default function PostForm() {
...
+
+```js
+const client = new GraphQLClient({
+  url: 'https://domain.com/graphql',
+  cache: memCache(),
```
Comment (bmullan91) [inline README.md]: We can remove this - it's not required for this example and may confuse people.

File: README.md [hunk 4]
```diff
@@ -583,6 +584,31 @@ export default function PostForm() {
...
+  fetch: require('node-fetch'),
+  FormData: require('formdata-node')
+})
+
+const uploadPostPictureMutation = /* GraphQL */ `
```
Comment (bmullan91) [inline README.md]: The editor hint `/* GraphQL */` is nice! For consistency could you add it to other queries in the docs or remove it from this one.

File: packages/graphql-hooks/test/unit/GraphQLClient.test.js [hunk 5]
```diff
@@ -345,16 +350,28 @@ describe('GraphQLClient', () => {
...
         ]
         expect(actual).toEqual(expected)
       })
 
       it('does not set Content-Type header', () => {
         expect(fetchOptions.headers).not.toHaveProperty('Content-Type')
       })
+
+      it('throws if no FormData polyfill provided', () => {
+        delete global.FormData
```
Comment (bmullan91) [inline packages/graphql-hooks/test/unit/GraphQLClient.test.js]: Could we replace it with an empty mock, and then re-instate the implementation after the test.

File: README.md (hunk 3 above)
Comment (lynxtaa) [reply README.md]: Ok, done

File: packages/graphql-hooks/test/unit/GraphQLClient.test.js (hunk 5 above)
Comment (lynxtaa) [reply packages/graphql-hooks/test/unit/GraphQLClient.test.js]: I've decided it's better to split the tests for browser and Node JS upload. Browser test is now using `window.FormData` and "real" file instance.

**Comment by bmullan91** [comment]
@all-contributors please add @lynxtaa for code test doc bug

**Comment by bmullan91** [comment]
@lynxtaa available in `graphql-hooks@4.3.0` 🚀

**Pull request merged**